File: src/algorithms/trees/bst-operations/bst-from-sorted-array/bst-from-sorted-array_test.py
import importlib
import sys
import os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "sources"))
module = importlib.import_module("bst-from-sorted-array")
BSTNode = module.BSTNode
bst_from_sorted_array = module.bst_from_sorted_array


def test_root_at_mid_value():
    result = bst_from_sorted_array([1, 2, 3, 4, 5, 6, 7])
    assert result.value == 4


def test_single_element():
    result = bst_from_sorted_array([42])
    assert result.value == 42
    assert result.left is None
    assert result.right is None


def test_empty_array():
    assert bst_from_sorted_array([]) is None


def test_two_element_tree():
    result = bst_from_sorted_array([1, 2])
    assert result.value == 1
    assert result.right.value == 2


def test_five_element_tree():
    result = bst_from_sorted_array([1, 2, 3, 4, 5])
    assert result.value == 3
    assert result.left.value == 1
    assert result.right.value == 4


if __name__ == "__main__":
    test_root_at_mid_value()
    test_single_element()
    test_empty_array()
    test_two_element_tree()
    test_five_element_tree()
    print("All tests passed!")

File: src/algorithms/trees/bst-operations/bst-from-sorted-array/__tests__/bst-from-sorted-array_test.rs
include!("../sources/bst-from-sorted-array.rs");

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_root_at_mid_value() {
        let result = bst_from_sorted_array(&[1, 2, 3, 4, 5, 6, 7]).unwrap();
        assert_eq!(result.value, 4);
    }

    #[test]
    fn test_single_element() {
        let result = bst_from_sorted_array(&[42]).unwrap();
        assert_eq!(result.value, 42);
        assert!(result.left.is_none());
        assert!(result.right.is_none());
    }

    #[test]
    fn test_empty_array() {
        assert!(bst_from_sorted_array(&[]).is_none());
    }

    #[test]
    fn test_two_element_tree() {
        let result = bst_from_sorted_array(&[1, 2]).unwrap();
        assert_eq!(result.value, 1);
        assert_eq!(result.right.as_ref().unwrap().value, 2);
    }

    #[test]
    fn test_five_element_tree() {
        let result = bst_from_sorted_array(&[1, 2, 3, 4, 5]).unwrap();
        assert_eq!(result.value, 3);
        assert_eq!(result.left.as_ref().unwrap().value, 1);
        assert_eq!(result.right.as_ref().unwrap().value, 4);
    }
}
